core/src/dunit/scala/io/snappydata/cluster/CassandraSnappyDUnitTest.scala

```diff
@@ -314,6 +326,161 @@ class CassandraSnappyDUnitTest(val s: String)
       rs.close()
     }
     assert(count == 0)
+  }
+  
+  def snap_2772BugTest_deployJar_createExternalTable(): Unit = {
+    logInfo("Running SNAP_2772 Bug Test 2")
+    val user1Conn = getConnection(netPort)
+    val stmt1 = user1Conn.createStatement()
+    stmt1.execute(s"deploy jar cassJar '$cassandraConnectorJarLoc'")
+    stmt1.execute("drop table if exists customer3")
+    stmt1.execute("create external table customer3 using org.apache.spark.sql.cassandra options" +
+        " (table 'customer', keyspace 'test', spark.cassandra.input.fetch.size_in_rows '200000'," +
+        " spark.cassandra.read.timeout_ms '10000')")
+    stmt1.execute("select * from customer3")
+    var rs = stmt1.getResultSet
+    var count = 0
+    if (rs ne null) {
+      while (rs.next()) {
+        count += 1
+      }
+      rs.close()
+    }
+    assert(count == 3)
 
+    stmt1.execute("list packages")
+    rs = stmt1.getResultSet
+    count = 0
+    if (rs ne null) {
+      while (rs.next()) {
+        count += 1
+      }
+      rs.close()
+    }
+    assert(count == 1)
+    stmt1.execute("undeploy cassJar")
+    stmt1.execute("list packages")
+    rs = stmt1.getResultSet
+    count = 0
+    if (rs ne null) {
+      while (rs.next()) {
+        count += 1
+      }
+      rs.close()
+    }
+    assert(count == 0)
+    try {
+      stmt1.execute("create external table customer3 using org.apache.spark.sql.cassandra options" +
+          " (table 'customer', keyspace 'test', " +
+          "spark.cassandra.input.fetch.size_in_rows '200000'," +
+          " spark.cassandra.read.timeout_ms '10000')")
+      assert(assertion = false, s"Expected an exception!")
+    } catch {
+      case _: SQLException => // expected
```

Comment:
Can do stricter checking with message text or SQLState code that comes in the SQLException